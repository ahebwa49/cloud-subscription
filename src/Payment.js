import React, { useState } from "react";
import "./Payment.scss";

const Payment = () => {
  const [cardNumber, setCardNumber] = useState("");
  const [expirationDate, setExpirationDate] = useState("2021-05-31");
  const [securityCode, setSecurityCode] = useState("");

  const onCardNumberChange = (e) => {
    console.log(e.target.value);
    setCardNumber(e.target.value);
  };

  const onExpirationDateChange = (e) => {
    console.log(e.target.value);
    setExpirationDate(e.target.value);
  };

  const onSecurityCodeChange = (e) => {
    console.log(e.target.value);
    setSecurityCode(e.target.value);
  };
  return (
    <div className="Payment">
      <h3>Payment Data </h3>
      <form className="SubscriptionForm">
        <div className="SubscriptionFormGroup">
          <label htmlFor="cardNumber" className="SubscriptionFormGroup__label">
            Card Number
          </label>
          <input
            type="text"
            name="cardNumber"
            placeholder="Card Number"
            required
            value={cardNumber}
            onChange={onCardNumberChange}
          />
        </div>
        <div className="SubscriptionFormGroup">
          <label
            htmlFor="expirationDate"
            className="SubscriptionFormGroup__label"
          >
            Expiration Date
          </label>
          <input
            type="date"
            required
            name="expirationDate"
            placeholder="Expiration Date"
            value={expirationDate}
            onChange={onExpirationDateChange}
          />
        </div>
        <div className="SubscriptionFormGroup">
          <label
            htmlFor="securityCode"
            className="SubscriptionFormGroup__label"
          >
            Security Code
          </label>
          <input
            type="text"
            name="securityCode"
            placeholder="Security Code"
            required
            value={securityCode}
            onChange={onSecurityCodeChange}
          />
        </div>
      </form>
    </div>
  );
};
export default Payment;
