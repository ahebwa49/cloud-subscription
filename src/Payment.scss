.Payment {
  display: flex;
  flex-direction: column;
  margin-top: 200px;
  padding: 1rem;
  width: 450px;
  height: 200px;
  border-radius: 4px;
  border: 1px solid #dadada;
}
