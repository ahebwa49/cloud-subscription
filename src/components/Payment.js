import React from "react";
import "../styles/Payment.scss";
import { calculateTotal } from "../utils/index";

const Payment = ({
  onCardNumberChange,
  onExpirationDateChange,
  onSecurityCodeChange,
  cardNumber,
  expirationDate,
  securityCode,
  selectedPlan,
  upfrontPayment,
  error,
  storage,
}) => {
  let totalPrice = calculateTotal(
    storage,
    selectedPlan.price_usd_per_gb,
    upfrontPayment
  );
  return (
    <div className="Payment">
      <h4>Payment Data </h4>
      <div className="PaymentPlan">
        Subscription Plan: {selectedPlan.duration_months} months,{" "}
        {selectedPlan.price_usd_per_gb}USD per GB
      </div>
      <div className="PaymentTotal">
        Amount of gigabytes in a cloud: {storage}GB
      </div>
      <div className="PaymentTotal">Upfront Payment: {upfrontPayment}</div>
      <div className="PaymentTotal">Total Price: {totalPrice}USD</div>
      <form className="PaymentForm">
        <div className="PaymentFormGroup">
          <label htmlFor="cardNumber" className="PaymentFormGroup__label">
            Card Number
          </label>
          <input
            type="text"
            name="cardNumber"
            placeholder="Card Number"
            required
            value={cardNumber}
            onChange={(e) => onCardNumberChange(e)}
          />
        </div>
        <div className="PaymentFormGroup">
          <label htmlFor="expirationDate" className="PaymentFormGroup__label">
            Expiration Date
          </label>
          <input
            type="date"
            required
            name="expirationDate"
            placeholder="Expiration Date"
            value={expirationDate}
            onChange={(e) => onExpirationDateChange(e)}
          />
        </div>
        <div className="PaymentFormGroup">
          <label htmlFor="securityCode" className="PaymentFormGroup__label">
            Security Code
          </label>
          <input
            type="text"
            name="securityCode"
            placeholder="Security Code"
            required
            value={securityCode}
            onChange={(e) => onSecurityCodeChange(e)}
          />
        </div>
      </form>
      {error ? <div className="paymentError">{error}</div> : null}
    </div>
  );
};
export default Payment;
