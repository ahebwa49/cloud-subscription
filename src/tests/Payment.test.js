import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Payment from "../components/Payment";

describe("create Payment snapshot", () => {
  test("create snapshot", () => {
    let tree = renderer
      .create(
        <Payment
          onCardNumberChange={() => {}}
          onExpirationDateChange={() => {}}
          onSecurityCodeChange={() => {}}
          cardNumber={1234}
          expirationDate={"2021-06-30"}
          securityCode={1234}
          selectedPlan={{ duration_months: 12, price_usd_per_gb: 2 }}
          storage={10}
          upfrontPayment="yes"
          error="All fields must be completed inorder to proceed to next step"
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
