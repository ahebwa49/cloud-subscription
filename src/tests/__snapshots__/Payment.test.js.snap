// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create Payment snapshot create snapshot 1`] = `
<div
  className="Payment"
>
  <h4>
    Payment Data 
  </h4>
  <div
    className="PaymentPlan"
  >
    Subscription Plan: 
    12
     months,
     
    2
    USD per GB
  </div>
  <div
    className="PaymentTotal"
  >
    Amount of gigabytes in a cloud: 
    10
    GB
  </div>
  <div
    className="PaymentTotal"
  >
    Upfront Payment: 
    yes
  </div>
  <div
    className="PaymentTotal"
  >
    Total Price: 
    20
    USD
  </div>
  <form
    className="PaymentForm"
  >
    <div
      className="PaymentFormGroup"
    >
      <label
        className="PaymentFormGroup__label"
        htmlFor="cardNumber"
      >
        Card Number
      </label>
      <input
        name="cardNumber"
        onChange={[Function]}
        placeholder="Card Number"
        required={true}
        type="text"
        value={1234}
      />
    </div>
    <div
      className="PaymentFormGroup"
    >
      <label
        className="PaymentFormGroup__label"
        htmlFor="expirationDate"
      >
        Expiration Date
      </label>
      <input
        name="expirationDate"
        onChange={[Function]}
        placeholder="Expiration Date"
        required={true}
        type="date"
        value="2021-06-30"
      />
    </div>
    <div
      className="PaymentFormGroup"
    >
      <label
        className="PaymentFormGroup__label"
        htmlFor="securityCode"
      >
        Security Code
      </label>
      <input
        name="securityCode"
        onChange={[Function]}
        placeholder="Security Code"
        required={true}
        type="text"
        value={1234}
      />
    </div>
  </form>
  <div
    className="paymentError"
  >
    All fields must be completed inorder to proceed to next step
  </div>
</div>
`;
